### ISSUE TITLE
extra_context does not work with private variables anymore

### ISSUE THREAD

**Issue by alexandersm** [cookiecutter/cookiecutter#1905]
* Cookiecutter version: 2.2.1
* Python version: 3.11.2
* Operating System: Debian GNU/Linux 12 (bookworm)

### Description:

I have my cookiecutter project when I define internal hidden variables and then assign values to them via extra_context parameter. 
Example:
```
{
  "_cookiecutter_version" :"1.0",
   "_extra": {},
}
```

Python code:
```
cookiecutter(
    './cookiecutter/1.0',
    extra_context={'timestamp': datetime.utcnow().isoformat(), "_extra": parsed_yaml}
)
```

I had no problem with Cookiecutter  version 2.1.1, _extra has all required elements from parsed_yaml variable

After the update to the version 2.2.1 it does not work anymore and **_extra** contains just empty OrderedDict()

Cookiecutter  2.2.1 is now able to use extra_context only with variables which are not private,which is not convenient, because non-private variables require user input 
```
{
  "_cookiecutter_version" :"1.0",
   "extra": {},
}
```

Can you please propose some workaround for that ? 
Thanks in advance !

**Comment by henryiii** [comment]
Hidden variables have two underscores, one underscore is a cookiecutter config variable. Have you tried two underscores?

**Comment by alexandersm** [comment]
No, haven't tried yet, but according to the doc at https://cookiecutter.readthedocs.io/en/stable/advanced/private_variables.html

> Cookiecutter allows the definition private variables by prepending an underscore to the variable name. The user will not be required to fill those variables in. These can either be not rendered, by using a prepending underscore, or rendered, prepending a double underscore.  

So I used single underscore to avoid rendering.
